### PR TITLE
Cy cat#1479

### LINKED ISSUE #1479: [Analyzer]CyCat
## Name
CyCat

## Link
https://cycat.org/about/

## Type of analyzer
generic analyzer:
* if a UUID is submitted, perform UUID lookup
* else perform full text search and then UUID lookups of the results

## Why should we use it
Encyclopedia of cyber security resources

## Possible implementation
follow docs here:https://cycat.org/services/

### PR DESCRIPTION
**Pull request by g4ze**
closes #1479

# Description

add cycat
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [x] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [x] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [x] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [x] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [x] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results. 
    - [x] I have added that raw JSON sample to the `MockUpResponse` of the `_monkeypatch()` method. This serves us to provide a valid sample for testing.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.
- [x] After you had submitted the PR, if `DeepSource`, `Django Doctors` or other third-party linters have triggered any alerts during the CI checks, I have solved those alerts.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.
image
 {
                            "description": """Detects Execution via
                            SyncInvoke in CL_Invocation.ps1 module""",
                            "raw": """
                            author: oscd.community, Natalia Shornikova
                            date: 2020/10/14
                            description: Detects Execution via
                            SyncInvoke in CL_Invocation.ps1 module
                            detection:
                            condition: selection
                            selection:
                                EventID: 4104
                                ScriptBlockText|contains|all:
                                - CL_Invocation.ps1
                                - SyncInvoke
                            falsepositives:
                            - Unknown
                            id: 4cd29327-685a-460e-9dac-c3ab96e549dc
                            level: high
                            logsource:
                            product: windows
                            service: powershell
                            modified: 2021/05/21
                            references:
                            - https://twitter.com/bohops/status/948061991012327424
                            status: experimental
                            tags:
                            - attack.defense_evasion
                            - attack.t1216
                            title: Execution via CL_Invocation.ps1
                            """,
                            "sigma:id": "4cd29327-685a-460e-9dac-c3ab96e549dc",
                            "title": "Execution via CL_Invocation.ps1",
                            "_cycat_type": "Item",
                        },

### CODE REVIEW

**Comment by g4ze** [comment]
@mlodic  kindly check the wrapper.

**Comment by mlodic** [comment]
lgtm

**Comment by g4ze** [comment]
TLP amber?

**Comment by g4ze** [comment]
I think we are gtg here.

File: api_app/analyzers_manager/migrations/0090_analyzer_config_cycat.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+plugin = {
+    "python_module": {
+        "health_check_schedule": None,
+        "update_schedule": None,
+        "module": "cycat.CyCat",
+        "base_path": "api_app.analyzers_manager.observable_analyzers",
+    },
+    "name": "CyCat",
+    "description": "[CyCat](https://cycat.org/)  or the CYbersecurity Resource CATalogue aims at mapping and documenting, in a single formalism and catalogue available cybersecurity tools, rules, playbooks, processes and controls.",
```
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0090_analyzer_config_cycat.py:16]:
```suggestion
    "description": "[CyCat](https://cycat.org/) or the CYbersecurity Resource CATalogue aims at mapping and documenting, in a single formalism and catalogue available cybersecurity tools, rules, playbooks, processes and controls.",
```

Comment (mlodic) [reply api_app/analyzers_manager/migrations/0090_analyzer_config_cycat.py:16]: :P

File: api_app/analyzers_manager/observable_analyzers/cycat.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    This analyzer is a wrapper for cycat api.
+    """
+
+    def update(self) -> bool:
+        pass
+
+    url: str = "https://api.cycat.org"
+
+    def uuid_lookup(self, uuid: str):
+        logger.info(f"performing lookup on uuid: {uuid}")
```
Comment (mlodic) [inline api_app/analyzers_manager/observable_analyzers/cycat.py:23]: adding the observable name in this log can help future debugging

File: api_app/analyzers_manager/observable_analyzers/cycat.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockUpResponse(
+                        [
+                            "24bfaeba-cb0d-4525-b3dc-507c77ecec41",
```
Comment (mlodic) [inline api_app/analyzers_manager/observable_analyzers/cycat.py:59]: I am not sure this is the expected response. I would have expected data, like the one here: https://cycat.org/services/#fetch-item-by-uuid

UUID are useless as an output so I would have expected to get the data for the UUID. Your analyzer seems right to me so I don't understand why you get this result. can you please try again?

**Review by mlodic — CHANGES_REQUESTED** [review]

File: api_app/analyzers_manager/observable_analyzers/cycat.py (hunk 3 above)
Comment (g4ze) [reply api_app/analyzers_manager/observable_analyzers/cycat.py:59]: yes sure, makes sense.

**Review by mlodic — APPROVED** [review]

**Pull request merged**